api/core/number.go

```diff
@@ -35,26 +37,19 @@ const (
 	Uint64NumberKind
 )
 
+var ErrInvalidNumberKindError = fmt.Errorf("Invalid number kind")
+
+// OutputFloatingPointPrecision determines the maximum precision of
+// any floating point number encoded by this package.
+const OutputFloatingPointPrecision = 16
+
 // Number represents either an integral or a floating point value. It
 // needs to be accompanied with a source of NumberKind that describes
 // the actual type of the value stored within Number.
 type Number uint64
 
 // - constructors
 
-// NewZeroNumber
-func NewZeroNumber(kind NumberKind) Number {
```

Comment:
Could you add a test that makes sure that `Number(0).As{Float,Int,Uint}64() == 0`?